docs/framework/interop/marshaling-classes-structures-and-unions.md

```diff
@@ -216,6 +217,69 @@ The `NativeMethods` class contains the prototypes for the `TestUnion` and `TestU
 [!code-csharp[Conceptual.Interop.Marshaling#29](~/samples/snippets/csharp/VS_Snippets_CLR/conceptual.interop.marshaling/cs/unions.cs#29)]
 [!code-vb[Conceptual.Interop.Marshaling#29](~/samples/snippets/visualbasic/VS_Snippets_CLR/conceptual.interop.marshaling/vb/unions.vb#29)]
 
+## Platform sample
+
+In some scenarios `struct` and `union` layouts can be different depending on the targeted platform. For example, consider the [`STRRET`](https://docs.microsoft.com/windows/win32/api/shtypes/ns-shtypes-strret) type when defined in a COM scenario:
+
+```c++
+#include <pshpack8.h> /* Defines the packing of the struct */
+typedef struct _STRRET
+    {
+    UINT uType;
+    /* [switch_is][switch_type] */ union
+        {
+        /* [case()][string] */ LPWSTR pOleStr;
+        /* [case()] */ UINT uOffset;
+        /* [case()] */ char cStr[ 260 ];
+        }  DUMMYUNIONNAME;
+    }  STRRET;
+#include <poppack.h>
+```
+
+The above `struct` is being declared with Windows' headers that influence the memory layout of the type. When defined in a managed environment these layout details are needed to properly interoperate with native code.
+
+The correct managed definition of this type in a 32-bit process would be:
+
+``` CSharp
+[StructLayout(LayoutKind.Explicit, Size = 264)]
+public struct STRRET_32
+{
+    [FieldOffset(0)]
+    public uint uType;
+
+    [FieldOffset(4)]
+    public IntPtr pOleStr;
+
+    [FieldOffset(4)]
+    public uint uOffset;
+
+    [FieldOffset(4)]
+    public IntPtr cStr;
+}
+```
+
+Where as on a 64-bit process, the correct layout would be:
+
+``` CSharp
+[StructLayout(LayoutKind.Explicit, Size = 272)]
+public struct STRRET_64
+{
+    [FieldOffset(0)]
+    public uint uType;
+
+    [FieldOffset(8)]
+    public IntPtr pOleStr;
+
+    [FieldOffset(8)]
+    public uint uOffset;
+
+    [FieldOffset(8)]
+    public IntPtr cStr;
+}
+```
```

Comment:
It would be nice to also add snippet here that shows the use, e.g. `if (IntPtr.Size == 8) { ... } else { ... }`